### ISSUE TITLE
Model unable to take input from dictionary(with input layer names as key) after loading it using tf.keras.models.load_model

### ISSUE THREAD

**Issue by Shubham3101** [tensorflow/tensorflow#32285]
**System information**
- OS Platform and Distribution (Linux Ubuntu 16.04):
device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0 rc
- Python version: 3.6

My model works fine the first time when i create the architecture and train it, but once i save the model using **tf.keras.model.save_model**  and then load it again using **tf.keras.models.load_model** it fails to take input from the  dictionary(with input layer names as key) as if the names of layers are changed.

Example:
training_mask = tf.keras.layers.Input(shape=(None, None, 1), name='training_mask')
target_score_map = tf.keras.layers.Input(shape=(None, None, 1), name='target_score_map')
target_geo_map = tf.keras.layers.Input(shape = (None,None,5), name='target_geo_map')

These are three input layers that i am using to take input for training but once i save the model and later load it again to train it gives me the error " No data provided for "input_1". Need data for each key in: ['input_1', 'input_2', 'input_3']"

I am feeding data by specifying layer names :

        inputs = {
                'target_score_map': score_maps,
                'target_geo_map': geo_maps,
                'training_mask': training_masks
            }

Therefore, it works the first time when create I the architecture from code then train and save it but later when i load the model from disk it fails.

Regards
Shubham

**Comment by ravikyram** [comment]
In order to expedite the trouble-shooting process, please provide a minimal standalone code to reproduce the issue reported here. Thanks!

**Comment by Shubham3101** [comment]
Hey ravikyram,

On more trouble-shooting i found out that the name of the layers are not changing, but something else is going on, layers are connected.

Following is the code to reproduce the issue:


**import tensorflow as tf
import os
import numpy as np
os.environ["CUDA_VISIBLE_DEVICES"] = "0"

def get_data():
    images = np.zeros((64,224))
    labels = np.zeros((64,5))
    inputs = {
        'Image_input':images
    }
    outputs = {
        'output-softmax':labels
    }
    return inputs, outputs

def create_model():
    input_layer = tf.keras.layers.Input(name='Image_input', shape=(224), dtype='float32')
    
    model = tf.keras.layers.Dense(5)(input_layer)
    model = tf.keras.layers.Activation('softmax', name = "output-softmax")(model)
    
    model = tf.keras.models.Model(inputs=input_layer, outputs=[model])
    return model
    
model = create_model()

optimizer = tf.keras.optimizers.Adadelta(lr=1.0, rho=0.95, epsilon=None, decay=0.0)
# print (model.summary())
model.compile(loss='categorical_crossentropy', optimizer=optimizer)
    

data = get_data()
model.fit(data[0], data[1], batch_size=16, epochs=10)

tf.keras.models.save_model(model,"model")

model1 = tf.keras.models.load_model("model")

model1.fit(data[0], data[1], batch_size=16, epochs=10)**

model.fit() works well but model1.fit() fails.
Also if you provide input without dictionary
**def get_data():
    images = np.zeros((64,224))
    labels = np.zeros((64,5))
    return images,labels**

both **model.fit()** and **model1.fit()** runs successfully.

Thanks

**Comment by ravikyram** [comment]
I have tried on colab with TF version 2.0 rc0 and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/d76d69451de64646e2917bb0b555e56c/untitled170.ipynb).Thanks!

**Comment by k-w-w** [comment]
I believe that this is fixed with commit: https://github.com/tensorflow/tensorflow/commit/6a26d679113105820cf83a2447863a6a32488c47#diff-29e6348914dee326daf67d86aa5075db.

Ran the colab (thanks @ravikyram!) with tf-nightly-2.0-preview, and it worked.

**Comment by Shubham3101** [comment]
Hello Everyone

I am still facing this issue with stable release of tensorflow-2 i.e tf-2.0.0.

Also, I am unable to use "model.summary()" after loading the model. It gives error:

**You tried to call `count_params` on Image_input, but the layer isn't built. You can build it manually via: `Image_input.build(batch_input_shape)`**

**Comment by Shubham3101** [comment]
@k-w-w Can you reopen this issue?

**Comment by jvishnuvardhan** [comment]
@Shubham3101 As mentioned by @k-w-w this was already resolved in `tf-nightly`. Please check the [gist here](https://colab.sandbox.google.com/gist/jvishnuvardhan/89df5873b5a8fcf9a5214a354235525d/tf_32285.ipynb). Thanks!

**Comment by Shubham3101** [comment]
Thanks @jvishnuvardhan, it is working with tf-nightly-2.1.0 version.
